src/data/shoprows.txt

```diff
@@ -1489,24 +1489,24 @@
 1488	conc	mini kiwi icepick	KIWI
 1489	conc	mini kiwi-flavored aspirin-injecting lipstick	KIWI
 1490	conc	mini kiwi digitized cookies	KIWI
-1491	row	bacteria bisque	The Primordial Soup Kitchen
-1492	row	ciliophora chowder	The Primordial Soup Kitchen
-1493	row	cream of chloroplasts	The Primordial Soup Kitchen
-1494	row	lip soup	The Primordial Soup Kitchen
-1495	row	lip soup	The Primordial Soup Kitchen
-1496	row	lip soup	The Primordial Soup Kitchen
-1497	row	elbow soup	The Primordial Soup Kitchen
-1498	row	elbow soup	The Primordial Soup Kitchen
-1499	row	elbow soup	The Primordial Soup Kitchen
-1500	row	flagellate soup	The Primordial Soup Kitchen
-1501	row	flagellate soup	The Primordial Soup Kitchen
-1502	row	flagellate soup	The Primordial Soup Kitchen
-1503	row	muscular soup	The Primordial Soup Kitchen
-1504	row	muscular soup	The Primordial Soup Kitchen
-1505	row	muscular soup	The Primordial Soup Kitchen
-1506	row	synaptic soup	The Primordial Soup Kitchen
-1507	row	synaptic soup	The Primordial Soup Kitchen
-1508	row	synaptic soup	The Primordial Soup Kitchen
+1491
```

Comment:
a mistaken removal? because the coinmaster doesn't exist?

Reply:
I moved the generation of the ShopRow logging from parsing coinmasters.txt to CoinmasterRegistry, after all the Coinmaster classes have been initialized.

That has two effects:

- We know what the "currencies" are and can accurately derive the ShopRow object even for "sell" records
- If there is ShopRow info in coinmasters.txt which is not used, it is missing.

I think a solution is to log "row" ShopRow data at parse time, and "buy" and "sell" data at registry time.

I'll try that out.

Reply:
Yeah. That worked like a charm. The Primordial Soup Kitchen entries are now in both shoprows.txt and allshoprows.txt